### PR TITLE
fix(article):   `Article` add `pending` `error` state  

### PR DESCRIPTION
**Pull request by gary02**
- add new states to  `Article`
- correct  article records state in database
- turn off most logging output in local testing

### CODE REVIEW

File: db/migrations/20230129101723_fix_article_state.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+exports.up = async (knex) => {
+  await knex.raw(/*sql*/ `
+  UPDATE article SET state = 'error' WHERE id IN (
+    SELECT article.id FROM draft 
+    LEFT JOIN article ON article.draft_id = draft.id
+      WHERE draft.publish_state != 'published' AND  article.draft_id IS NOT NULL
```
Comment (robertu7) [inline db/migrations/20230129101723_fix_article_state.js:6]: all drafts that `publish_state` in `unpublished`, `error`, and `pending` are included?

File: src/connectors/queue/__test__/publication.test.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+describe('publicationQueue.publishArticle', () => {
+  test('publish not pending draft', async () => {
+    const notPendingDraftId = '1'
+    const draft = await publicationQueue.draftService.baseFindById(
+      notPendingDraftId
+    )
+    expect(draft.state).not.toBe(PUBLISH_STATE.pending)
+
+    const job = await publicationQueue.publishArticle({
```
Comment (robertu7) [inline src/connectors/queue/__test__/publication.test.ts:14]: aha, now I know how to test an async queue job!

File: src/connectors/queue/publication.ts [hunk 3]
```diff
@@ -361,6 +337,11 @@ class PublicationQueue extends BaseQueue {
         )
       }
 
+      await this.articleService.baseUpdate(article.id, {
```
Comment (robertu7) [inline src/connectors/queue/publication.ts:340]: shall we move [update `draft.publishState` to `published`](https://github.com/thematters/matters-server/blob/b95b0de1c1d79986a7d9ef1e72e6797f0c596a9b/src/connectors/queue/publication.ts#L141) to here?

**Review by robertu7 — APPROVED** [review]

File: db/migrations/20230129101723_fix_article_state.js (hunk 1 above)
Comment (gary02) [reply db/migrations/20230129101723_fix_article_state.js:6]: Yes, this is what the sql does.  You mean `draft.publish_state != 'published'` should be `draft.publish_state = 'error'` ?

**Pull request merged**